src/dirs.proj

```diff
@@ -17,6 +17,9 @@
     <Project Include="XMakeTasks\UnitTests\Microsoft.Build.Tasks.UnitTests.csproj" />
     <Project Include="XMakeBuildEngine\UnitTests\Microsoft.Build.Engine.UnitTests.csproj" />
     <Project Include="XMakeBuildEngine\UnitTestsPublicOM\Microsoft.Build.Engine.OM.UnitTests.csproj" />
+
+    <Project Include="Test\Projects\Dependency\Dependency.csproj" />
+    <Project Include="Test\Projects\TaskWithDependency\TaskWithDependency.csproj" />
```

Comment:
Both, unfortunately.

From what I can tell about msbuild's "convention" regarding test projects:
- Samples are like big, end 2 end smoke tests. But they are also examples that show how to do certain things. They are like big stakes in the ground. They really are not exhaustive :).
- Each test class defines its own convention on test samples. Sadly. Some just have a csproj as a literal string. Some construct entire test projects on the fly in some temp directory. When I first added tests to `Typeloader_Tests` to test task dll loading under xplat, I chose to reuse the PortableTask sample. In your case, it feels that the test project that you added might be of use as a reference for other people wanting to create tasks with dependencies, so putting it in Samples makes some sense.